core/embed/trezorhal/stm32u5/secret.c

```diff
@@ -173,11 +177,19 @@ secbool secret_optiga_extract(uint8_t *dest) {
     }
 
     for (int j = 0; j < 4; j++) {
-      dest[i * 4 + j] = (val >> (j * 8)) & 0xFF;
+      secret_ptr[i * 4 + j] = (val >> (j * 8)) & 0xFF;
     }
   }
 
-  return all_zero ? secfalse : sectrue;
+  if (all_zero) {
+    return secfalse;
+  }
```

Comment:
well these have different function

`secret_present` (`secret_optiga_present`) check whether the secret is written in flash, which used for decision whether to hide the Hide Protect area so that firmware/prodtest can't access it, modify it etc.

while 

`secret_optiga_extract` (now `secret_optiga_get`) look into the backup registers (on U5) to check whether the secret was provided by bootloader. Which will not be the case if the firmware is unofficial.

The empty values are different for backup registers and flash.

Reply:
I see. Let's document this in the header files, because I think this is not at all obvious from the function names. Namely:
- What are the preconditions for each function to be used:
    - What needs to be done before the function can be called (some other function needs to have written the value to the registers, for example).
    - When can the functions be meaningfully called (bootloader vs. official firmware vs. unofficial firmware)
- What does the return value of each of these functions mean.

Reply:
Added comments in https://github.com/trezor/trezor-firmware/commit/8d4fce5129e3172718e1a2cefe514d7cab5d5e97

Note: i will do a bit of refactoring of the secret.h api, to hide more platform differences inside, similarly to what you suggested with `secret_optiga_set` and `secret_optiga_get`. In order to simplify the secret handling here: https://github.com/trezor/trezor-firmware/blob/285b22e53e880fda171eb05ace67fa7a94b8e145/core/embed/bootloader/main.c#L328

I will do that in successive PR if you don't mind.

Reply:
Also we should make the other function names more intuitive. In case of `secret_hide()` vs. `secret_optiga_hide()`, it should be more obvious that the second clears the register, whereas the first restricts access to the flash. That's totally different stuff. For example `secret_optiga_backup()` -> `secret_optiga_cache()` and `secret_optiga_hide` -> `secret_optiga_uncache()`, or `_load` vs. `_unload`. Same applies to `secret_bhk_provision()` -> `secret_bhk_cache()` or `secret_bhk_load()` depending on whatever `secret_optiga_backup()` will be renamed to.